src/bytes_mut.rs

```diff
@@ -391,8 +393,11 @@ impl BytesMut {
 
         unsafe {
             let mut other = self.shallow_clone();
-            other.set_end(at);
-            self.set_start(at);
+            other.cap = at;
+            other.len = at;
+            // SAFETY: We've checked that `at` <= `self.len()` and we know that `self.len()` <=
+            // `self.capacity()`.
+            self.advance_unchecked(at);
```

Comment:
Very minor nitpick, but this is unsafe code so I think it might be worth applying. In `split_off`, you call `advance_unchecked` first then update `cap` and `len`. You reverse it here. Would you mind keeping them the same?

Reply:
Yeah, thanks for pointing it out! Made things a little more consistent in b97174340bad78f2df33d8fb8bef9c841c6e511b. Let me know if that isn't what you meant.